client/rpc/src/state/mod.rs

```diff
@@ -360,10 +379,98 @@ impl<Block, Client> State<Block, Client>
 			}.boxed()
 		})?;
 
+		module.register_subscription(
+			"state_runtimeVersion",
+			"state_unsubscribeRuntimeVersion",
+			|_params, mut sink, ctx| {
+				let executor = ctx.executor.clone();
+				let client = ctx.client.clone();
+
+				let mut previous_version = client.runtime_version_at(&BlockId::hash(client.info().best_hash))
+					.expect("best hash is valid; qed");
+				let _ = sink.send(&previous_version);
+				let rt_version_stream = client.storage_changes_notification_stream(Some(&[StorageKey(well_known_keys::CODE.to_vec())]), None, )
+					.map_err(|blockchain_err| Error::Client(Box::new(blockchain_err)))
+					.map_err(to_jsonrpsee_call_error)?;
+
+				let fut = async move {
+					let mut stream = rt_version_stream
+						.filter_map(move |_| {
+							let info = client.info();
+							let version = client
+								.runtime_version_at(&BlockId::hash(info.best_hash));
+								match version {
+									Ok(v) => if previous_version != v {
+											previous_version = v.clone();
+											future::ready(Some(v))
+										} else {
+											future::ready(None)
+										},
+									Err(e) => {
+										log::error!("Could not fetch current runtime version. Error={:?}", e);
+										// TODO: Does this terminate the stream? What is the best way to let users know?
+										future::ready(None)
+									}
+								}
+						});
+						stream.for_each(|version| {
+							match sink.send(&version) {
+								Ok(_) =>  future::ready(()),
+								Err(e) => {
```

Comment:
that's fine but you could also just bail if you get `Error::SubscriptionClosed` maybe `stream.take_while` is better.